journals/apps/journals/models.py

```diff
@@ -330,8 +330,14 @@ def update_service(client, data, service_name):
             except HttpNotFoundError as err:
                 # Only a WARN because this will often happen on JournalAboutPage creation.
                 logging.warn(f"JournalAboutPage unable to update {service_name} because UUID doesn't exist: {err.content}")
+            except HttpServerError as err:
```

Comment:
If the UUID doesn't exist, we should be getting a `HttpNotFoundError` not a `HttpServerError`. What's the error your getting that's causing you to catch this?

Reply:
@tuchfarber 
I expected that as well, but was getting 500 errors when the uuid didn't exist and I don't understand why. Here is the stack trace from ecommerce:
```
2018-05-01 18:37:35,451 ERROR 12 [django.request] /edx/app/ecommerce/venvs/ecommerce/local/lib/python2.7/site-packages/django/core/handlers/exception.py:135 - Internal Server Error: /journal/api/v1/journals/c127b712-b2b8-4c8a-9455-8c7da4792a2d/
edx.devstack.ecommerce | Traceback (most recent call last):
edx.devstack.ecommerce |   File "/edx/app/ecommerce/venvs/ecommerce/local/lib/python2.7/site-packages/django/core/handlers/base.py", line 131, in get_response
edx.devstack.ecommerce |     response = middleware_method(request, response)
edx.devstack.ecommerce |   File "/edx/app/ecommerce/venvs/ecommerce/local/lib/python2.7/site-packages/django/contrib/flatpages/middleware.py", line 12, in process_response
edx.devstack.ecommerce |     return flatpage(request, request.path_info)
edx.devstack.ecommerce |   File "/edx/app/ecommerce/venvs/ecommerce/local/lib/python2.7/site-packages/django/contrib/flatpages/views.py", line 35, in flatpage
edx.devstack.ecommerce |     site_id = get_current_site(request).id
edx.devstack.ecommerce |   File "/edx/app/ecommerce/venvs/ecommerce/local/lib/python2.7/site-packages/django/contrib/sites/shortcuts.py", line 15, in get_current_site
edx.devstack.ecommerce |     return Site.objects.get_current(request)
edx.devstack.ecommerce |   File "/edx/app/ecommerce/venvs/ecommerce/local/lib/python2.7/site-packages/django/contrib/sites/models.py", line 65, in get_current
edx.devstack.ecommerce |     return self._get_site_by_request(request)
edx.devstack.ecommerce |   File "/edx/app/ecommerce/venvs/ecommerce/local/lib/python2.7/site-packages/django/contrib/sites/models.py", line 50, in _get_site_by_request
edx.devstack.ecommerce |     SITE_CACHE[domain] = self.get(domain__iexact=domain)
edx.devstack.ecommerce |   File "/edx/app/ecommerce/venvs/ecommerce/local/lib/python2.7/site-packages/django/db/models/manager.py", line 85, in manager_method
edx.devstack.ecommerce |     return getattr(self.get_queryset(), name)(*args, **kwargs)
edx.devstack.ecommerce |   File "/edx/app/ecommerce/venvs/ecommerce/local/lib/python2.7/site-packages/django/db/models/query.py", line 380, in get
edx.devstack.ecommerce |     self.model._meta.object_name
edx.devstack.ecommerce | DoesNotExist: Site matching query does not exist.
edx.devstack.ecommerce | 2018-05-01 18:37:35,646 ERROR 12 [django.server] /edx/app/ecommerce/venvs/ecommerce/local/lib/python2.7/site-packages/django/core/servers/basehttp.py:124 - "PATCH /journal/api/v1/journals/c127b712-b2b8-4c8a-9455-8c7da4792a2d/ HTTP/1.1" 500 130162
edx.devstack.ecommerce | 2018-05-01 18:37:35,646 ERROR 12 [django.server] /edx/app/ecommerce/venvs/ecommerce/local/lib/python2.7/site-packages/django/core/servers/basehttp.py:124 - "PATCH /journal/api/v1/journals/c127b712-b2b8-4c8a-9455-8c7da4792a2d/ HTTP/1.1" 500 130162
edx.devstack.ecommerce | 2018-05-01 18:37:35,814 INFO 12 [django.server] /edx/app/ecommerce/venvs/ecommerce/local/lib/python2.7/site-packages/django/core/servers/basehttp.py:124 - "POST /journal/api/v1/journals/ HTTP/1.1" 201 332
edx.devstack.ecommerce | 2018-05-01 18:37:35,814 INFO 12 [django.server] /edx/app/ecommerce/venvs/ecommerce/local/lib/python2.7/site-packages/django/core/servers/basehttp.py:124 - "POST /journal/api/v1/journals/ HTTP/1.1" 201 332
```

Reply:
@tuchfarber 
If I do a get on the endpoint for an unknown uuid I get a notfounderror. Not sure why the patch call is causing this

Reply:
That's because you don't have edx.devstack.ecommerce in your Sites in the ecommerce IDA. I don't know why it only fails sometimes, but if you swap your localhost:18000 to edx.devstack.ecommece:18000 this error will go away. It will also cause the err.content to be correct.

Reply:
awesome, thanks!
confirmed that now we get the HttpNotFoundError so I removed catching server error